src/toolbar_gui.cpp

```diff
@@ -450,13 +450,25 @@ static CallBackFunction MenuClickMap(int index)
 
 /* --- Town button menu --- */
 
+enum TownMenuEntries {
+	TME_SHOW_DIRECTORY = 0,
+	TME_SHOW_FOUND_TOWN,
+	TME_SHOW_PLACE_HOUSES,
+};
+
 static CallBackFunction ToolbarTownClick(Window *w)
 {
-	if (_settings_game.economy.found_town == TF_FORBIDDEN) {
-		PopupMainToolbarMenu(w, WID_TN_TOWNS, {STR_TOWN_MENU_TOWN_DIRECTORY});
-	} else {
-		PopupMainToolbarMenu(w, WID_TN_TOWNS, {STR_TOWN_MENU_TOWN_DIRECTORY, STR_TOWN_MENU_FOUND_TOWN});
-	}
+	DropDownList list;
+	list.push_back(MakeDropDownListStringItem(STR_TOWN_MENU_TOWN_DIRECTORY, TME_SHOW_DIRECTORY));
+
+	/* Maybe allow founding towns. */
```

Comment:
```suggestion
	/* Add option for "founding towns" if setting is enabled. */
```
Or something like that. But reading `Maybe` before an `if` feels very off :)

I also wonder if this actually needs a comment, but that I leave up to you :)